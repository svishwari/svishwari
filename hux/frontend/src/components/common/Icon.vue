<template>
  <svg-as-component
    :src="`assets/icons/${type}`"
    :width="enableCustomSize ? customSize.width : size"
    :height="enableCustomSize ? customSize.height : size"
    :color="color"
    :variant="variant"
    :fill-opacity="fillOpacity"
    :stroke="stroke"
    :stroke-opacity="strokeOpacity"
  />
</template>

<script>
import svgAsComponent from "./SVG"

export default {
  name: "Icon",

  components: {
    svgAsComponent,
  },

  props: {
    type: {
      type: String,
      required: true,
    },

    size: {
      type: [Number, String],
      required: false,
      default: 60,
    },

    enableCustomSize: {
      type: Boolean,
      required: false,
      default: false,
    },

    customSize: {
      type: Object,
      required: false,
      default() {
        return {
          width: 0,
          height: 0,
        }
      },
    },

    color: {
      type: String,
      required: false,
    },

    variant: {
      type: String,
      required: false,
    },

    stroke: {
      type: String,
      required: false,
    },

    fillOpacity: {
      type: Number,
      required: false,
      default: 1,
    },

    strokeOpacity: {
      type: Number,
      required: false,
      default: 1,
    },
  },
}
</script>
