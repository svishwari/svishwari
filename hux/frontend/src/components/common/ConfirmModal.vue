<template>
  <v-dialog v-model="localModal" :width="width">
    <template #activator="{ on, attrs }">
      <slot name="activator" v-bind="attrs" v-on="on"></slot>
    </template>
    <template #default>
      <div class="confirm-modal-wrapper">
        <div class="confirm-modal-body px-6">
          <slot name="icon">
            <icon v-if="icon" :type="icon" :color="localIconColor" :size="42" />
          </slot>
          <slot name="title">
            <div v-if="title" class="black--text text--darken-4 text-h3 pt-3">
              {{ title }}
            </div>
          </slot>
          <slot name="sub-title">
            <div
              v-if="subTitle"
              class="black--text text--darken-4 text-h3 mt-n2"
            >
              {{ subTitle }}
            </div>
          </slot>
          <slot name="body">
            <div v-if="body" class="black--text text--darken-4 text-h6 pt-6">
              {{ body }}
            </div>
          </slot>
        </div>
        <div class="confirm-modal-footer">
          <slot name="footer">
            <huxButton
              size="large"
              variant="white"
              height="40"
              is-tile
              @click="onCancel()"
            >
              <span class="primary--text">{{ leftBtnText }}</span>
            </huxButton>
            <huxButton
              size="large"
              :variant="type"
              height="40"
              is-tile
              @click="onConfirm()"
            >
              {{ rightBtnText }}
            </huxButton>
          </slot>
        </div>
      </div>
    </template>
  </v-dialog>
</template>

<script>
import huxButton from "@/components/common/huxButton"
import Icon from "@/components/common/Icon.vue"
export default {
  name: "ConfirmModal",

  components: {
    huxButton,
    Icon,
  },

  props: {
    icon: {
      type: String,
      required: false,
    },

    iconColor: {
      type: String,
      required: false,
    },

    type: {
      type: String,
      required: false,
      default: "primary",
    },

    title: {
      type: String,
      required: false,
    },

    subTitle: {
      type: String,
      required: false,
    },

    body: {
      type: String,
      required: false,
    },

    leftBtnText: {
      type: String,
      required: false,
      default: "Nevermind!",
    },

    rightBtnText: {
      type: String,
      required: false,
      default: "Add",
    },

    value: {
      type: Boolean,
      required: true,
      default: false,
    },

    width: {
      type: Number,
      required: false,
      default: 600,
    },
  },

  data() {
    return {
      localModal: this.value,
    }
  },

  computed: {
    localIconColor() {
      if (this.iconColor) {
        return this.iconColor
      }
      return this.type
    },
  },

  watch: {
    value: function () {
      this.localModal = this.value
    },

    localModal: function () {
      this.$emit("input", this.localModal)
      if (!this.localModal) {
        this.$emit("onClose")
      }
    },
  },

  methods: {
    onCancel: function () {
      this.$emit("onCancel")
    },
    onConfirm: function () {
      this.$emit("onConfirm")
    },
  },
}
</script>

<style lang="scss" scoped>
.confirm-modal-wrapper {
  background: var(--v-white-base);
  text-align: center;
  padding-top: 42px;
  .confirm-modal-footer {
    border-top: 1px solid var(--v-black-lighten3);
    margin-top: 36px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 20px 40px;
    background: var(--v-primary-lighten1);
  }
}
</style>
