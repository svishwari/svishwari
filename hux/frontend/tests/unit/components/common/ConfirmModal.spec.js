import ConfirmModal from "@/components/common/ConfirmModal"
import Icon from "@/components/common/Icon"
import { shallowMount } from "@vue/test-utils"

describe("Tests Confirm Modal", () => {
  test("Confirm modal calls function on close", async () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        value: true,
        icon: "sad-face",
      },
    })
    await wrapper.setData({ localModal: false })
    expect(typeof wrapper.emitted().onClose === "function")
  })

  test("Confirm modal displays props content", () => {
    const props = {
      value: true,
      title: "Title prop",
      rightBtnText: "Button prop",
      body: "Body prop",
      subTitle: "Subtitle prop",
      icon: "sad-face",
    }
    const wrapper = shallowMount(ConfirmModal, {
      propsData: props,
    })

    expect(wrapper.text()).toContain(props.title)
    expect(wrapper.text()).toContain(props.rightBtnText)
    expect(wrapper.text()).toContain(props.body)
    expect(wrapper.text()).toContain(props.subTitle)
    expect(wrapper).toMatchSnapshot()
  })

  test("Confirm modal displays slots content", () => {
    const bodyContent = "Body slot"
    const titleSlotContent = "Title slot"
    const footerSlotContent =
      "This is a footer slot that displays at the bottom"
    const subTitleSlotContent =
      "This is a subtitle slot that displays below the title"

    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        value: true,
        icon: "sad-face",
      },
      slots: {
        body: `<div>${bodyContent}</div>`,
        title: `<div>${titleSlotContent}</div>`,
        footer: `<div>${footerSlotContent}</div>`,
        "sub-title": `<div>${subTitleSlotContent}</div>`,
      },
    })
    expect(wrapper.text()).toContain(bodyContent)
    expect(wrapper.text()).toContain(titleSlotContent)
    expect(wrapper.text()).toContain(footerSlotContent)
    expect(wrapper.text()).toContain(subTitleSlotContent)
    expect(wrapper).toMatchSnapshot()
  })
  test("Confirm modal displays icon content", () => {
    const wrapper = shallowMount(ConfirmModal, {
      propsData: {
        value: true,
        icon: "sad-face",
      },
      stubs: {
        icon: Icon,
      },
    })

    expect(wrapper).toMatchSnapshot()
  })
})
