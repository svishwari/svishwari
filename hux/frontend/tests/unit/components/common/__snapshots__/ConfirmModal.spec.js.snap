// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Confirm Modal Confirm modal displays icon content 1`] = `
<v-dialog-stub opendelay="0" closedelay="0" value="true" contentclass="" maxwidth="none" origin="center center" retainfocus="true" transition="dialog-transition" width="600">
  <div class="confirm-modal-wrapper">
    <div class="confirm-modal-body px-6">
      <svg-as-component-stub src="assets/icons/sad-face" width="42" height="42" color="primary" variant="base" fillopacity="1" strokeopacity="1"></svg-as-component-stub>
      <!---->
      <!---->
      <!---->
    </div>
    <div class="confirm-modal-footer">
      <huxbutton-stub variant="white" istile="true" size="large" height="40" boxshadow="true" iconcolor="primary" iconvariant="base" iconsize="14" iconclass="ml-2 mr-2" class="btn-border box-shadow-none"><span class="primary--text">Nevermind!</span></huxbutton-stub>
      <huxbutton-stub variant="primary" istile="true" size="large" height="40" boxshadow="true" iconcolor="primary" iconvariant="base" iconsize="14" iconclass="ml-2 mr-2">
        Add
      </huxbutton-stub>
    </div>
  </div>
</v-dialog-stub>
`;

exports[`Tests Confirm Modal Confirm modal displays props content 1`] = `
<v-dialog-stub opendelay="0" closedelay="0" value="true" contentclass="" maxwidth="none" origin="center center" retainfocus="true" transition="dialog-transition" width="600">
  <div class="confirm-modal-wrapper">
    <div class="confirm-modal-body px-6">
      <icon-stub type="sad-face" size="42" color="primary" fillopacity="1" strokeopacity="1"></icon-stub>
      <div class="black--text text--darken-4 text-h2 pt-3">
        Title prop
      </div>
      <div class="black--text text--darken-4 text-h2 mt-n2">
        Subtitle prop
      </div>
      <div class="
              black--text
              text--darken-4 text-subtitle-1
              pt-6
              font-weight-regular
            ">
        Body prop
      </div>
    </div>
    <div class="confirm-modal-footer">
      <huxbutton-stub variant="white" istile="true" size="large" height="40" boxshadow="true" iconcolor="primary" iconvariant="base" iconsize="14" iconclass="ml-2 mr-2" class="btn-border box-shadow-none"><span class="primary--text">Nevermind!</span></huxbutton-stub>
      <huxbutton-stub variant="primary" istile="true" size="large" height="40" boxshadow="true" iconcolor="primary" iconvariant="base" iconsize="14" iconclass="ml-2 mr-2">
        Button prop
      </huxbutton-stub>
    </div>
  </div>
</v-dialog-stub>
`;

exports[`Tests Confirm Modal Confirm modal displays slots content 1`] = `
<v-dialog-stub opendelay="0" closedelay="0" value="true" contentclass="" maxwidth="none" origin="center center" retainfocus="true" transition="dialog-transition" width="600">
  <div class="confirm-modal-wrapper">
    <div class="confirm-modal-body px-6">
      <icon-stub type="sad-face" size="42" color="primary" fillopacity="1" strokeopacity="1"></icon-stub>
      <div>Title slot</div>
      <div>This is a subtitle slot that displays below the title</div>
      <div>Body slot</div>
    </div>
    <div class="confirm-modal-footer">
      <div>This is a footer slot that displays at the bottom</div>
    </div>
  </div>
</v-dialog-stub>
`;
